File: clojure/bob/src/bob.clj
(ns bob)

(defn response-for [say] 
	(def say-trim (clojure.string/trim say))
	(if (= "" say-trim) (str "Fine. Be that way!")
		(if (and (not= nil (re-matches #"(.*)[a-zA-Z](.*)" say-trim))
			  (= (clojure.string/upper-case say-trim) say-trim))
			(str "Whoa, chill out!")
			(if (clojure.string/ends-with?  say-trim "?") 
				(str "Sure.")
				(str "Whatever."))))
)

File: python/bob/bob.py
def hey(phrase):
    phrase = phrase.strip()
    if phrase == "":
        return "Fine. Be that way!"
    asking = phrase.endswith("?")
    yelling = phrase == phrase.upper() and (phrase.isupper() or phrase.islower())
    if asking and yelling:
        return "Calm down, I know what I'm doing!"
    if asking:
        return "Sure."
    if yelling:
        return "Whoa, chill out!"
    else:
        return "Whatever."
